legal-api/src/legal_api/resources/nr_type_map.py

```diff
@@ -0,0 +1,197 @@
+# Copyright © 2021 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Name Request Type mapping service.
+
+Provides an endpoint to query/match NR Type to Entity Type mappings.
+"""
+from enum import Enum
+from http import HTTPStatus
+
+from flask import request
+from flask_restx import Namespace, Resource
+
+from legal_api.models import Business
+
+
+API = Namespace('nrTypeMap', description='NR Type mapping service')
+
+
+class RequestActions(Enum):
+    """Render an Enum of the Request Actions. Should match RequestAction enum in namex project."""
+
+    AML = 'AML'  # amalgamate
+    AS = 'AS'  # assumed
+    CHG = 'CHG'  # change (alter)
+    CNV = 'CNV'  # convert
+    MVE = 'MVE'  # move
+    NEW = 'NEW'  # new business
+    NEW_AML = 'NEW_AML'  # new or amalgamate
+    REH = 'REH'
+    REN = 'REN'  # renew
+    REST = 'REST'  # restore
+
+
+@API.route('', methods=['GET'])
+@API.doc(params={
+    'nrType': {'description': 'Optional NR Type code.', 'example': 'CR'},
+    'entityType': {'description': 'Optional Entity Type code.', 'example': 'BC'},
+    'RequestActions': {'description': 'Optional Request Action code.', 'example': 'NEW'}
+})
+class NrTypeMap(Resource):
+    """NR Type mapping service."""
+
+    @staticmethod
+    def get():
+        """Return a list of mappings matching any query params."""
+        # get query params
+        nr_type_cd = (request.args.get('nrTypeCd', None).upper()) \
+            if request.args.get('nrTypeCd', None) else None
+        entity_type_cd = (request.args.get('entityTypeCd', None).upper()) \
+            if request.args.get('entityTypeCd', None) else None
+        request_action_cd = (request.args.get('requestActionCd', None).upper()) \
+            if request.args.get('requestActionCd', None) else None
+
+        # build list of tuples
+        # should match Request Type Mapping in namex project
+        request_type_mapping = [
```

Comment:
suggest this be a dict, gives a O(1) instead of an O(n)

Reply:
I wanted to do this but:
- there is no unique key
- I wanted to allow querying by any column (eg, `GET /api/v1/nrTypeMap?nrTypeCd=CR`

Thoughts?